ivy/functional/ivy/gradients.py

```diff
@@ -37,17 +37,43 @@ def __exit__(self, exc_type, exc_val, exc_tb):
 # Gradient Mode #
 
 # noinspection PyShadowingNames
-def with_grads(with_grads=None):
-    """Summary.
+def with_grads(with_grads: bool = None) -> bool:
+    """
+    Enter a nested code space where gradients are computed. This method
+    adds the with_grads component to the global list with_grads_stack
 
     Parameters
     ----------
     with_grads
-         (Default value = None)
+        Boolean value denoting whether the current code block has gradient
```

Comment:
where is the code block? If it corresponds to code blocks, why are there not code blocks in the example?